src/Storages/VirtualColumnUtils.h

```diff
@@ -74,12 +77,13 @@ struct VirtualsForFileLikeStorage
     std::optional<size_t> size { std::nullopt };
     const String * filename { nullptr };
     std::optional<Poco::Timestamp> last_modified { std::nullopt };
-
 };
 
+std::map<std::string, std::string> parseFromPath(const std::string& path);
+
 void addRequestedFileLikeStorageVirtualsToChunk(
     Chunk & chunk, const NamesAndTypesList & requested_virtual_columns,
-    VirtualsForFileLikeStorage virtual_values);
+    VirtualsForFileLikeStorage virtual_values, const std::string & hive_partitioning_path = "");
```

Comment:
Let's add `hive_partitioning_path` argument to `VirtualsForFileLikeStorage` structure